### PR TITLE
Marked serde-failed messages with red

### PR DESCRIPTION
**Pull request by David-DB88**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
Marked serde-failed messages with red in Topics messages 
**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx [hunk 1]
```diff
@@ -115,11 +118,13 @@ const Message: React.FC<Props> = ({
           <div>{formatTimestamp(timestamp)}</div>
         </td>
         <StyledDataCell title={key}>
+          <S.Icon>{keySerde === 'Fallback' && <WarningRedIcon />}</S.Icon>
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:121]: do we have an `enum` for this ? `'Fallback'`

Comment (David-DB88) [reply kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:121]: i didn't find

**Review by Mgrdich — APPROVED** [review]

**Review by Haarolean — CHANGES_REQUESTED** [review]
1. there should be no tooltip
image
2. the icon should be on the right side of the cell  
image

**Comment by David-DB88** [comment]
@Haarolean i do not added tooltip i don't no why it shows to you, it doesn't shows for me.

**Review by Haarolean — APPROVED** [review]

**Review by workshur — APPROVED** [review]

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx [hunk 2]
```diff
@@ -18,7 +19,15 @@ const StyledDataCell = styled.td`
...
-
+const StyledIsLongText = styled.div`
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  max-width: 340px;
+`;
+const StyledDataCellFlex = styled.div`
+  display: flex;
+`;
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:30]: Move it to styled.ts

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx [hunk 3]
```diff
@@ -18,7 +19,15 @@ const StyledDataCell = styled.td`
   max-width: 350px;
   min-width: 350px;
 `;
-
+const StyledIsLongText = styled.div`
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  max-width: 340px;
+`;
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:27]: I would suggest to create ellipsis common component with width property

File: kafka-ui-react-app/src/components/common/Icons/WarningRedIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+const WarningRedIcon: React.FC = () => {
+  return (
+    <svg
+      width="20"
+      height="20"
+      viewBox="0 0 20 20"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <rect width="20" height="20" rx="10" fill="#FAD1D1" />
```
Comment (workshur) [inline kafka-ui-react-app/src/components/common/Icons/WarningRedIcon.tsx:12]: Color should be moved to theme

File: kafka-ui-react-app/src/components/common/Icons/WarningRedIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      <path d="M9 4.74219H11V12.7422H9V4.74219Z" fill="#E61A1A" />
```
Comment (workshur) [inline kafka-ui-react-app/src/components/common/Icons/WarningRedIcon.tsx:13]: Color should be moved to theme

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx (hunk 2 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx]: done

File: kafka-ui-react-app/src/components/common/Icons/WarningRedIcon.tsx (hunk 4 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/common/Icons/WarningRedIcon.tsx]: done

File: kafka-ui-react-app/src/components/common/Icons/WarningRedIcon.tsx (hunk 5 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/common/Icons/WarningRedIcon.tsx]: done

**Review by Mgrdich — APPROVED** [review]

File: kafka-ui-react-app/src/components/common/Ellipsis/Ellipsis.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+};
+
+const Ellipsis: React.FC<PropsType> = ({ serde, children }) => {
+  return (
+    <S.StyledDataCellFlex>
+      <S.StyledIsLongText>{children}</S.StyledIsLongText>
+      <S.Icon>{serde === 'Fallback' && <WarningRedIcon />}</S.Icon>
+    </S.StyledDataCellFlex>
+  );
+};
```
Comment (workshur) [inline kafka-ui-react-app/src/components/common/Ellipsis/Ellipsis.tsx:18]:
```suggestion
const Ellipsis: React.FC<PropsWithChildren<{ text: string }>> = ({ text, children }) => {
  return (
    <S.Wrapper>
      <S.Text>{children}</S.Text>
      {children && (<S.Postfix>{children}</S.Postfix>)}
    </S.Wrapper>
  );
};
```

File: kafka-ui-react-app/src/components/common/Ellipsis/Ellipsis.styled.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  max-width: 340px;
+`;
+export const StyledDataCellFlex = styled.div`
+  display: flex;
+`;
+
+export const Icon = styled.span`
+  padding-left: 0.3rem;
+  vertical-align: top;
+`;
```
Comment (workshur) [inline kafka-ui-react-app/src/components/common/Ellipsis/Ellipsis.styled.ts:16]:
```suggestion
export const Wrapper = styled.div`
  display: flex;
  gap: 8px;
  align-items: center;
`;

```

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx [hunk 8]
```diff
@@ -116,16 +106,20 @@ const Message: React.FC<Props> = ({
...
+        <S.StyledDataCell title={key}>
+          <Ellipsis serde={keySerde}>
+            {renderFilteredJson(key, keyFilters)}
+          </Ellipsis>
+        </S.StyledDataCell>
+        <S.StyledDataCell title={content}>
           <S.Metadata>
             <S.MetadataValue>
-              {renderFilteredJson(content, contentFilters)}
+              <Ellipsis serde={valueSerde}>
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:117]: Ellipsis is a common component. why we sending serde property here?

**Review by workshur — CHANGES_REQUESTED** [review]

File: kafka-ui-react-app/src/theme/theme.ts [hunk 9]
```diff
@@ -551,6 +551,10 @@ const theme = {
     closeIcon: Colors.neutral[30],
     deleteIcon: Colors.red[20],
     warningIcon: Colors.yellow[20],
+    warningRedIcon: {
+      rectFill: Colors.red[10],
+      pathFill: '#E61A1A',
```
Comment (workshur) [inline kafka-ui-react-app/src/theme/theme.ts:556]: It must be color from `Colors` object

**Review by workshur — CHANGES_REQUESTED** [review]

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx [hunk 10]
```diff
@@ -116,16 +106,20 @@ const Message: React.FC<Props> = ({
...
           <div>{formatTimestamp(timestamp)}</div>
         </td>
-        <StyledDataCell title={key}>
-          {renderFilteredJson(key, keyFilters)}
-        </StyledDataCell>
-        <StyledDataCell title={content}>
+        <S.StyledDataCell title={key}>
+          <Ellipsis serde={keySerde}>
+            {renderFilteredJson(key, keyFilters)}
+          </Ellipsis>
```
[the remark this answers is not in the document]
Comment (Mgrdich) [reply kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:112]: i think something stayed in russian here

File: kafka-ui-react-app/src/theme/theme.ts (hunk 9 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/theme/theme.ts:556]: done

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx [hunk 11]
```diff
@@ -116,16 +106,20 @@ const Message: React.FC<Props> = ({
         <td>
           <div>{formatTimestamp(timestamp)}</div>
         </td>
-        <StyledDataCell title={key}>
-          {renderFilteredJson(key, keyFilters)}
-        </StyledDataCell>
-        <StyledDataCell title={content}>
+        <S.StyledDataCell title={key}>
```
[the remark this answers is not in the document]
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:109]: done

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx (hunk 8 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:117]: done

File: kafka-ui-react-app/src/components/common/Ellipsis/Ellipsis.styled.ts (hunk 7 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/common/Ellipsis/Ellipsis.styled.ts:16]: done

File: kafka-ui-react-app/src/components/common/Ellipsis/Ellipsis.tsx (hunk 6 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/common/Ellipsis/Ellipsis.tsx:18]: done

**Review by workshur — APPROVED** [review]
Code looks good to me

**Review by Mgrdich — APPROVED** [review]

**Comment by Haarolean** [comment]
@armenuikafka fixed

**Review by armenuikafka — APPROVED** [review]

**Pull request merged**